absl/status/status_payload_printer.cc

```diff
@@ -16,26 +16,27 @@
 #include <atomic>
 
 #include "absl/base/attributes.h"
+#include "absl/base/internal/atomic_hook.h"
 
 namespace absl {
 ABSL_NAMESPACE_BEGIN
 namespace status_internal {
 
 namespace {
-// Tried constant initialized global variable but it doesn't work with Lexan
-// (MSVC's `std::atomic` has trouble constant initializing).
-std::atomic<StatusPayloadPrinter>& GetStatusPayloadPrinterStorage() {
-  ABSL_CONST_INIT static std::atomic<StatusPayloadPrinter> instance{nullptr};
+absl::base_internal::AtomicHook<StatusPayloadPrinter>&
+GetStatusPayloadPrinterStorage() {
+  ABSL_CONST_INIT static absl::base_internal::AtomicHook<StatusPayloadPrinter>
```

Comment:
This actually needs to be a global, so you can remove the function that wraps it. It should also be `ABSL_INTERNAL_ATOMIC_HOOK_ATTRIBUTES`, not `ABSL_CONST_INIT`. Don't initialize it to `nullptr` either.

Reply:
@derekmauro Thank you for the comment. I've changed the commit to introduce the global variable.